### PR TITLE
README: add indexes section

### PR DESCRIPTION
**Pull request by mcarmonaa**
Signed-off-by: Manuel Carmona <manu.carmona90@gmail.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -174,6 +176,42 @@ To be able to create your own data source implementation you need to implement t
 
 You can see a really simple data source implementation on our `mem` package.
 
+## Indexes
+
+`go-mysql-server` exposes a series of interfaces to allow you to implement your own indexes so you can speedup your queries.
+
+Taking a look at the main index interface:
+
+```go
```
Comment (ajnavarro) [inline README.md]: do not copy the code here, use a github snippet.

File: README.md [hunk 2]
```diff
@@ -174,6 +176,42 @@ To be able to create your own data source implementation you need to implement t
...
+	ExpressionHashes() []ExpressionHash
+	// Driver ID of the index.
+	Driver() string
+}
+```
+
+You must note a couple of constraints:
+
+- this abstraction will index one or more columns.
+- if you want to index an expression you will only be able to index **one and only one** expression at a time.
```
Comment (ajnavarro) [inline README.md]: ...if you want to index an expression **that is not a column**...

Comment (kuba--) [reply README.md]: maybe:
> This abstraction lets you create an index for multiple columns (one or more) or for **only one** expression (e.g. function applied on multiple columns).

File: README.md [hunk 3]
```diff
@@ -174,6 +176,17 @@ To be able to create your own data source implementation you need to implement t
 
 You can see a really simple data source implementation on our `mem` package.
 
+## Indexes
+
+`go-mysql-server` exposes a series of interfaces to allow you to implement your own indexes so you can speedup your queries.
+
+Taking a look at the main [index interface](https://github.com/src-d/go-mysql-server/blob/0d36af3c06c4c200a20e6e6f7eb9e29045e98b81/sql/index.go#L35), you must note a couple of constraints:
```
Comment (ajnavarro) [inline README.md]: inside the go-mysql-server we don't want to put the specific hash, each documentation link will be pointing to the correct place.

File: README.md [hunk 4]
```diff
@@ -174,6 +176,17 @@ To be able to create your own data source implementation you need to implement t
...
+
+- This abstraction lets you create an index for multiple columns (one or more) or for **only one** expression (e.g. function applied on multiple columns).
+- If you want to index an expression that is not a column you will only be able to index **one and only one** expression at a time.
+
```
Comment (ajnavarro) [inline README.md]: why that spaces?

File: README.md (hunk 3 above)
Comment (ajnavarro) [reply README.md]: ohh, what I meant is only possible on issues markdown:

https://github.com/src-d/go-mysql-server/blob/0d36af3c06c4c200a20e6e6f7eb9e29045e98b81/sql/index.go#L35-L52

Anyway, If it is possible, we should point to master or godoc instead to a specific commit hash.

File: README.md (hunk 4 above)
Comment (mcarmonaa) [reply README.md]: ninja spaces, I didn't notice of them

**Pull request merged**